include/z64map.h

```diff
@@ -1,6 +1,140 @@
 #ifndef Z64MAP_H
 #define Z64MAP_H
+#include "ultra64.h"
+#include "z64scene.h"
+
+/*
+Handles L-minimap and pause screen dungeon minimap system.
+
+z_map_data.c stores MapSpriteInfo and helper functions to "simplify" accessing this data
+
+Two separate Id systems are used to access data
+
+mapId
+0x000-0x004 fetches L-minimap data from gameplay_dangeon_keep
+0x005-0x039 fetches dungeon minimap data from map_i_static
+0x100-0x161 fetches L-minimap data from map_grand_static
+
+mapCompactId
+0x000-0x039 fetches dungeon minimap data from map_i_static
+0x03A-0x09B fetches L-minimap data from map_grand_static
+*/
 
 #define FLOOR_INDEX_MAX 4
 
+/* z_map_data */
+#define MAPDATA_MAP_I_MAX 0x3A
+#define MAPDATA_MAP_GRAND 0x100
+#define MAPDATA_GET_MAP_GRAND_ID_FROM_COMPACT_ID(id) (id + (MAPDATA_MAP_GRAND - MAPDATA_MAP_I_MAX))
+#define MAPDATA_GET_MAP_GRAND_ID_FROM_MAP_ID(mapId) (mapId - MAPDATA_MAP_GRAND)
+
+#define MAPDATA_MID_GAMEPLAY_DANGEON_KEEP 0
+#define MAPDATA_MID_MAP_GRAND_STATIC 1
+#define MAPDATA_MID_MAP_I_STATIC 2
+
+#define MAPDATA_CPID_MAP_GRAND_STATIC 0
+#define MAPDATA_CPID_MAP_I_STATIC 1
+#define MAPDATA_CPID_2 2
```

Comment:
Could you walk me through a lot of abbreviations in here? What's `MAP_I`? `MID`? `CPID`?

Reply:
`MAP_I` refers to the `map_i_static` segment. This file stores the 58 (0x3A) textures to create pause menu dungeon maps.

`MID` should refer to the `mapId` indexing of minimaps, described in the header, used to fetch sprite data.

`CPID` should refer to the "compacted" mapId system (`mapCompactId`), also described in the header.

The `MAPDATA_MID` and `MAPDATA_CPID` enums are used to determine the source segment file of the texture

Reply:
I think we should spell out the full names instead, i.e `MAP_ID` instead of `MID`.
The current naming is confusing

Reply:
Not ignoring this, just prefer to hold off on this for a moment in case I think of a better naming system now that the map_disp is OK.